test/system/projects/samples_test.rb

```diff
@@ -247,10 +247,43 @@ class SamplesTest < ApplicationSystemTestCase
 
       assert_selector 'a', text: I18n.t('projects.samples.index.new_button'), count: 0
       assert_selector 'h1', text: I18n.t('projects.samples.index.title')
-      assert_selector 'table#samples-table tr', count: 2
+      assert_selector 'table#samples-table tr', count: 3
       assert_selector 'table#samples-table tr button.Viral-Dropdown--icon', text: '', count: 0
       assert_text @sample1.name
       assert_text @sample2.name
     end
+
+    test 'can search the list of samples by name' do
+      visit namespace_project_samples_url(namespace_id: @namespace.path, project_id: @project.path)
+
+      assert_selector 'table#samples-table tr', count: 3
+      assert_text @sample1.name
+      assert_text @sample2.name
+
+      fill_in I18n.t(:'projects.samples.index.search.placeholder'), with: samples(:sample1).name
+
+      assert_selector 'table#samples-table tr', count: 2
+      assert_text @sample1.name
+      assert_no_text @sample2.name
+    end
+
+    test 'can sort the list of projects' do
+      visit namespace_project_samples_url(namespace_id: @namespace.path, project_id: @project.path)
+
+      assert_selector 'table#samples-table tr', count: 3
+      within first('tr[class="text-sm font-normal text-slate-500 dark:text-slate-400"]') do
+        assert_text @sample1.name
+      end
+
+      click_on I18n.t(:'projects.samples.index.sorting.updated_at_desc')
+      click_on I18n.t(:'projects.samples.index.sorting.name_desc')
+      sleep 1
+      click_on I18n.t(:'projects.samples.index.sorting.name_desc')
```

Comment:
I was just wondering why the descending name sort needs to be clicked twice? Can we remove one?

Reply:
Thanks, this should've been an assert_text with the filter name on it. Fixed in 767377956c07734fd18058edd4b8dc9ac1af1da1